### PR TITLE
[FE-116] [FE-117] [FE-118] Add, edit, delete points in student profile

### PR DESCRIPTION
**Pull request by AnCichocka**
https://oops2000.atlassian.net/browse/SCRUM-116
https://oops2000.atlassian.net/browse/SCRUM-117
https://oops2000.atlassian.net/browse/SCRUM-118

@Szyntos please double check if teacher without edit rights can add, edit and delete points - I think I manage to perform mutation without rights?

There are some issues that needs to be corrected for example:
- handling unexpected state in form,
- clearing form error,
which I would like to correct in the follow up ticket, if you find anything sus - let me know.

### CODE REVIEW

**Comment by AnCichocka** [comment]
@Szyntos I would like you to test this out - I faced some backend errors

File: frontend/src/graphql/formCategories.graphql [hunk 1]
```diff
@@ -1,15 +1,16 @@
-query Categories($editionId: bigint!) {
+query FormCategories($editionId: bigint!) {
```
Comment (Szyntos) [inline frontend/src/graphql/formCategories.graphql:1]:
```graphql
query FormCategories {
  categories(orderBy: {categoryName: ASC}, where: {categoryEditions: {editionId: {_eq: 1}}}) {
    categoryId
    categoryName
    subcategories(orderBy: {ordinalNumber: ASC}, where: {editionId: {_eq: 1}}) {
      subcategoryId
    }
    canAddPoints
  }
}
```


You should also check the category's edition

File: frontend/src/screens/StudentProfile/TeacherStudentProfile.tsx [hunk 2]
```diff
@@ -72,15 +83,68 @@ export function TeacherStudentProfile() {
...
+            categories={formCategories}
+            handleConfirmClick={handleEditPointsConfirmation}
+            mutationError={editPointsError?.message}
+            initialValues={{
+              subcategoryId: pointsToEdit?.subcategory.subcategoryId as string,
+              points: pointsToEdit?.points.purePoints?.value ?? 0,
+              categoryId: pointsToEdit?.subcategory.category
+                .categoryId as string,
+            }}
+            variant="edit"
```
Comment (Szyntos) [inline frontend/src/screens/StudentProfile/TeacherStudentProfile.tsx:109]: imo the text on all of the buttons should be "Confirm"

**Review by Szyntos — CHANGES_REQUESTED** [review]
Small change, but overall big brain PR

File: frontend/src/screens/StudentProfile/TeacherStudentProfile.tsx (hunk 2 above)
Comment (AnCichocka) [reply frontend/src/screens/StudentProfile/TeacherStudentProfile.tsx:109]: I don't think so, I believe CTA button should describe what it does - add or edit, but we can discuss it further

File: frontend/src/graphql/formCategories.graphql (hunk 1 above)
Comment (AnCichocka) [reply frontend/src/graphql/formCategories.graphql:1]: done

**Review by Szyntos — APPROVED** [review]

**Pull request merged**